src/IO/Resources/AnimationsLoader.cs

```diff
@@ -90,6 +90,61 @@ public void CreateAtlas(GraphicsDevice device)
             new List<Tuple<ushort, byte>>(), new List<Tuple<ushort, byte>>()
         };
 
+        /* Load from the given file at the given index into the slot at body */
+        private void LoadMulBody(ushort body, ushort index, ANIMATION_GROUPS_TYPE type, UOFile idxFile)
+        {
+            long offsetToData = DataIndex[body].CalculateOffset(index, type, out int count);
+            long maxaddress = idxFile.StartAddress.ToInt64() + idxFile.Length;
+
+            if (offsetToData >= idxFile.Length)
+            {
+                return;
+            }
+
+            DataIndex[index].Type = type;
+
+            bool isValid = false;
+            long address = idxFile.StartAddress.ToInt64() + offsetToData;
+
+            DataIndex[body].Groups = new AnimationGroup[100];
+
+            int offset = 0;
+
+            for (byte j = 0; j < 100; j++)
+            {
+                DataIndex[body].Groups[j] = new AnimationGroup
+                {
+                    Direction = new AnimationDirection[5]
+                };
+
+                if (j >= count)
+                {
+                    continue;
+                }
+
+                for (byte d = 0; d < 5; d++)
+                {
+                    if (DataIndex[body].Groups[j].Direction[d] == null)
+                    {
+                        DataIndex[body].Groups[j].Direction[d] = new AnimationDirection();
+                    }
+
+                    AnimIdxBlock* aidx = (AnimIdxBlock*)(address + offset * sizeof(AnimIdxBlock));
+                    ++offset;
+
+                    if ((long)aidx < maxaddress && aidx->Size != 0 && aidx->Position != 0xFFFFFFFF && aidx->Size != 0xFFFFFFFF)
+                    {
+                        DataIndex[body].Groups[j].Direction[d].Address = aidx->Position;
+                        DataIndex[body].Groups[j].Direction[d].Size = aidx->Size;
+
+                        isValid = true;
+                    }
```

Comment:
for `bodyconv.def` the code must be this:

```csharp
if ((long)aidx < maxaddress && aidx->Position != 0xFFFFFFFF && aidx->Size != 0xFFFFFFFF)
{
    ref var direction = ref DataIndex[index].Groups[j].Direction[d];
  
    direction.Address = aidx->Position;
    direction.Size = Math.Max(1, aidx->Size);
  
    isValid = true;
}                                     
```

In particular the `aidx->Size != 0` must be removed. Time ago i found an animation with size == 0 but with 1 valid frame.

Reply:
just found the commit:
https://github.com/ClassicUO/ClassicUO/commit/4eb3f9f6e6e9e67765ab28e1e7adfa9bdee551da

Reply:
Ok. I didn't mean to change this logic so I'll fix it up.